modules/rpc/yarpc.go

```diff
@@ -186,6 +284,38 @@ func RegisterDispatcher(dispatchFn yarpcDispatcherFn) {
 	_dispatcherFn = dispatchFn
 }
 
-func defaultYarpcDispatcher(_ service.Host, cfg yarpc.Config) (*yarpc.Dispatcher, error) {
+func defaultYARPCDispatcher(_ service.Host, cfg yarpc.Config) (*yarpc.Dispatcher, error) {
 	return yarpc.NewDispatcher(cfg), nil
 }
+
+// Dispatcher returns a dispatcher that can be used to create clients.
+// It should be called after all modules have been started, because
+// the each module start creates a new dispatcher and stops previous.
+func Dispatcher() *yarpc.Dispatcher {
+	_dispatcherMu.RLock()
+	defer _dispatcherMu.RUnlock()
+	return _dispatcher
+}
+
+// Stops the current dispatcher, sets the new one with the host, config and _dispatcherFn, and then starts it.
+func resetDispatcher(host service.Host, config yarpc.Config) error {
+	d, err := _dispatcherFn(host, config)
+	if err != nil {
+		return err
+	}
+
+	_dispatcherMu.Lock()
+	defer _dispatcherMu.Unlock()
+
+	if _dispatcher == nil {
+		_dispatcher = d
+		return _dispatcher.Start()
+	}
+
+	if err := _dispatcher.Stop(); err != nil {
```

Comment:
I don't like the idea of stopping dispatcher and restarting new one. Can we do away with it somehow? You will be stopping the world for one module Stop() call. How about we start up new Dispatcher, free up the lock and move old dispatcher for closing with drain?